src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupRunner.kt

```diff
@@ -109,5 +130,331 @@ object RollupRunner : ScheduledJobRunner,
+                                    is RollupIndexResult.Success -> RollupResult.Success(compositeRes, rollupIndexResult.stats)
+                                    is RollupIndexResult.Failure -> RollupResult.Failure(rollupIndexResult.message, rollupIndexResult.cause)
+                                }
+                            }
+                            is RollupSearchResult.Failure -> {
+                                RollupResult.Failure(rollupSearchResult.message, rollupSearchResult.cause)
+                            }
+                        }
+                        when (rollupResult) {
+                            is RollupResult.Success -> {
+                                metadata = rollupMetadataService.updateMetadata(updatableJob,
+                                    metadata.mergeStats(rollupResult.stats), rollupResult.internalComposite)
+                                updatableJob = client.suspendUntil { listener: ActionListener<GetRollupResponse> ->
+                                    execute(GetRollupAction.INSTANCE, GetRollupRequest(updatableJob.id, null, "_local"), listener)
+                                }.rollup ?: throw IllegalStateException("Unable to get rollup job")
+                            }
+                            is RollupResult.Failure -> {
+                                rollupMetadataService.updateMetadata(
+                                    metadata.copy(status = RollupMetadata.Status.FAILED, failureReason = rollupResult.cause.message)
+                                )
+                            }
+                        }
+                        try {
+                            BackoffPolicy.exponentialBackoff(TimeValue.timeValueMillis(
+                                RollupSettings.DEFAULT_RENEW_LOCK_RETRY_DELAY),
+                                RollupSettings.DEFAULT_RENEW_LOCK_RETRY_COUNT
+                            ).retry(logger) {
+                                updatableLock = context.lockService.suspendUntil { renewLock(updatableLock, it) }
+                            }
+                        } catch (e: Exception) {
+                            logger.warn("Failed trying to renew lock on $updatableLock", e)
+                            // If we fail to renew the lock it doesn't mean we need to perm fail the job, we can just return early
+                            // and let the next execution try to process the data from where this one left off
+                            return
+                        }
+                    } catch (e: RollupMetadataException) {
+                        // Rethrow this exception so it doesn't get consumed here
+                        logger.info("RollupMetadataException being thrown", e)
+                        throw e
+                    } catch (e: Exception) {
+                        // TODO: Should update metadata and disable job here instead of allowing the rollup to keep going
```

Comment:
This TODO should also be a candidate for inclusion whenever we work on retry logic.